user/applications/tinker/src/burnin_test.cpp

```diff
@@ -459,5 +473,147 @@ bool BurninTest::testCpuLoad() {
+	// EXAMPLE:
+	// $<TalkerID>GGA,<UTC>,<Lat>,<N/S>,<Lon>,<E/W>,<Quality>,<NumSatUsed>,<HDOP>,<Alt>,M,<Sep>,M,<DiffAge>,<DiffStation>*<Checksum><CR><LF>
+	// +QGPSGNMEA: $GPGGA,213918.00,3804.405282,N,12209.922544,W,1,05,3.0,145.9,M,-26.4,M,,*5E
+
+    //Log.info("%s", buf);
+
+	*gnssLocked = false;
+
+	const int MAX_GPGGA_STR_LEN = 256;
+    char gpggaSentence[MAX_GPGGA_STR_LEN] = {};
+    strlcpy(gpggaSentence, buf, MAX_GPGGA_STR_LEN);
+
+    String lattitudeLongitude("LAT/LONG:");
+    int numberSattelites = 0;
+    
+    const char * delimiters = ",";
+    char * token = strtok(gpggaSentence, delimiters);
+    int i = 1;
+    while (token) {
+    	//Log.trace("%d %s", i, token);
+    	token = strtok(NULL, delimiters);
+    	i++;
+
+    	switch (i) {
+	    	case 3: // Lattitude or checksum if no lock and field is empty
+	    		if (strlen(token) > 5) {
+	    			lattitudeLongitude.concat(" ");
+	    			lattitudeLongitude.concat(token);
+	    		}
+	    		break;
+	    	case 4: // N/S
+	    	case 5: // Longitude
+	    	case 6: // E/W
+				lattitudeLongitude.concat(" ");
+    			lattitudeLongitude.concat(token);
+	    		break;
+	    	case 8: // Number satellites
+	    		numberSattelites = (int)String(token).toInt();
+	    		if (numberSattelites > 0) {
+	    			*gnssLocked = true;	
+    			    Log.info("%s Satellites: %d", lattitudeLongitude.c_str(), numberSattelites);
+	    		}
+	    		break;
+	    	default:
+	    		break;
+    	}
+    }
+
+    return 1;
+}
+
+bool BurninTest::testGnss() {
+	// Turn on GNSS + Modem
+    pinMode(GNSS_ANT_PWR, OUTPUT);
+	digitalWrite(GNSS_ANT_PWR, HIGH);
+
+	if (!turnModemOn()) {
+		return false;
+	}
+
+	// Enable GNSS. It can take some time after the modem AT interface comes up for the GNSS engine to start
+	const int RETRIES = 10;
+	int r = 0;
+	for (int i = 0; i < RETRIES && r != RESP_OK; i++) {
+		r = Cellular.command("AT+QGPS=1\r\n");
+		delay(1000);
+	}
+
+	if (r != RESP_OK) {
+		strcpy(BurninErrorMessage, "AT+QGPS=1 failed, GNSS not enabled");
+		return false;
+	}
+
+	// Configure antenna for GNSS priority
+	for (int i = 0; i < RETRIES && r != RESP_OK; i++) {
+		r = Cellular.command("AT+QGPSCFG=\"priority\",0");
+		delay(1000);
+	}
+
+	if (r != RESP_OK) {
+		strcpy(BurninErrorMessage, "AT+QGPSCFG=\"priority\",0 failed, GNSS not prioritized");
+		return false;
+	}
+
+	// Poll NMEA GGA sentence
+	// Parse for satellite lock bit, parse rough lat/long + print it
+	bool gnssLocked = false;
+	const int GNSS_POLL_TIMEOUT_MS = 90000;
+	auto timeout = millis() + GNSS_POLL_TIMEOUT_MS;
+	while (millis() < timeout && !gnssLocked) {
+		Cellular.command(callbackGPSGGA, &gnssLocked, 1000, "AT+QGPSGNMEA=\"GGA\"");
+		Log.info("gnssLocked %d", gnssLocked);
+		delay(1000);
+	}
+
+	// If no lock in X minutes, fail
+	if (!gnssLocked) {
+		strcpy(BurninErrorMessage, "No GNSS lock after 90s");
+	}
+
+	// Turn off Cell modem + GNSS antenna
+	digitalWrite(GNSS_ANT_PWR, LOW);
+	Cellular.off();
+	waitFor(Cellular.isOff, 30000);
```

Comment:
`turnModemOff()` ?  It could also not explicitly fail the test with `false` if it doesn't turn off.